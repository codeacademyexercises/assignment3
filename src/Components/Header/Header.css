.header {
    background-color: #494748;
    height: 20%;
    font-size: 32px;
    font-weight: 500;
    color: white;
    display: flex;
    flex-direction: column;
    justify-content:flex-start;
    align-items: center;
}

.header div{
    width: 100%;
}

.HorizontalLine {
    display: flex;
    /* flex-direction: row; */
    justify-content: flex-start;
}
.HorizontalLine hr{
    margin: 0px;
    background-color: white;
    color: white;
    height: 5px;
    flex-basis: 50%;
}

.HorizontalLine :last-child{
    /* align-self: flex-end; */
    margin-left: 15%; 
}