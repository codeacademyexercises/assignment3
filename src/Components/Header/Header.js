import React, { Component } from 'react';
import './Header.css';

class Header extends Component{
    render(){
        return(
            <header className="header">
                <span>BS</span>
                <div className="HorizontalLine">
                    <hr />
                    <hr />
                </div>
                <span>The Book Shelf</span>
            </header>
        );
    }
}

export default Header;