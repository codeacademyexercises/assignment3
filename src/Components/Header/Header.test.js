import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from './Header';

describe('<Header /> ',()=>{
    it('snapshot test',()=>{
        const wrapper = shallow(<Header />);
        expect(wrapper).toMatchSnapshot();
    })
})