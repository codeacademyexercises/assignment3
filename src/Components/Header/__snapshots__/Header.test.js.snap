// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header />  snapshot test 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          BS
        </span>,
        <div
          className="HorizontalLine"
        >
          <hr />
          <hr />
        </div>,
        <span>
          The Book Shelf
        </span>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "BS",
        },
        "ref": null,
        "rendered": "BS",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <hr />,
            <hr />,
          ],
          "className": "HorizontalLine",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "The Book Shelf",
        },
        "ref": null,
        "rendered": "The Book Shelf",
        "type": "span",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            BS
          </span>,
          <div
            className="HorizontalLine"
          >
            <hr />
            <hr />
          </div>,
          <span>
            The Book Shelf
          </span>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "BS",
          },
          "ref": null,
          "rendered": "BS",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <hr />,
              <hr />,
            ],
            "className": "HorizontalLine",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "hr",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "The Book Shelf",
          },
          "ref": null,
          "rendered": "The Book Shelf",
          "type": "span",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
